src/modules/olMap/components/OlMap.js

```diff
@@ -187,19 +188,23 @@ export class OlMap extends MvuElement {
 
 		//register particular obeservers on our Model
 		//handle fitRequest
-		this.observeModel('fitRequest', (fitRequest) => {
-			this._viewSyncBlocked = true;
+		this.observeModel(['fitRequest', 'fitLayerRequest'], (eventLike) => {
 			const onAfterFit = () => {
 				this._viewSyncBlocked = false;
 				this._syncStore();
 			};
-			const maxZoom = fitRequest.payload.options.maxZoom || this._view.getMaxZoom();
-			this._view.fit(fitRequest.payload.extent, { maxZoom: maxZoom, callback: onAfterFit });
+			const extent = eventLike.payload.id ? getLayerById(this._map, eventLike.payload.id).getSource()?.getExtent() : eventLike.payload.extent;
+
+			if (extent) {
+				this._viewSyncBlocked = true;
+				const maxZoom = eventLike.payload.options.maxZoom || this._view.getMaxZoom();
+				this._view.fit(extent, { maxZoom: maxZoom, callback: onAfterFit }); // Todo: fit map with padding parameter by calling mapService#calculatePadding
+			}
```

Comment:
Can we extract this to a helper-method, to be consistent to the code unterneath?

Reply:
>to be consistent to the code unterneath?

 Sorry, didn't get the point, can you explain it more in detail, please?

Reply:
sure...

Line 205 and 207 uses only simple calls to private functions,  the code in line 192 - 202 can be moved to a simple method, too. In the result, the reader gets a consistent overview over all observed properties and the related behavior.